test/unittests/test_cpp.cpp

```diff
@@ -120,3 +124,36 @@ TEST(cpp, host_call)
 
     example_host_destroy_context(host_context);
 }
+
+TEST(cpp, result_raii)
+{
+    static auto release_called = false;
+    release_called = false;
+    auto release_fn = [](const evmc_result*) noexcept { release_called = true; };
+
+    {
+        auto raw_result = evmc_result{};
+        raw_result.status_code = EVMC_INTERNAL_ERROR;
+        raw_result.release = release_fn;
+
+        auto raii_result = evmc::result{raw_result};
```

Comment:
Actually another question. When we wrap it here, shouldn't it set `raw_result.release` to `nullptr`?

Reply:
That might be good idea, but requires changing constructor from `const evmc_result&` to `evmc_result&&`. That requires at least another PR.

Reply:
Tracked in #285.